test/e2e/replica_set_change_version/replica_set_test.go

```diff
@@ -18,7 +19,15 @@ func TestMain(m *testing.M) {
 
 func TestReplicaSetUpgradeVersion(t *testing.T) {
 	ctx := f.NewContext(t)
-	defer ctx.Cleanup()
+
+	// When the tests are ran in the evg host we don't want to clean up
+	// the context, as we need it to dump diagnostics
+	// In all other cases (e.g. tests ran locally) we still clean up
+	// the context when we finish
+	if os.Getenv("SKIP_CLEANUP") != "1" {
+		defer ctx.Cleanup()
+	}
+
 	if err := e2eutil.RegisterTypesWithFramework(&mdbv1.MongoDB{}); err != nil {
```

Comment:
maybe we could incorporate the above check into this function, or create a separate `TestSetup` type function that handles this for us. It could be easy to forget to include this check.